irohad/torii/cache/cache.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_CACHE_HPP
+#define IROHA_CACHE_HPP
+
+#include <endpoint.pb.h>
+#include <boost/optional.hpp>
+#include <list>
+#include <mutex>
+#include <string>
+
+namespace torii {
+  namespace cache {
```

Comment:
With respect to issues below, may be better to rework namespace of cache and make it more general.